### PR TITLE
Updates to support metrics SSL

### PR DESCRIPTION
**Pull request by gp510**

### CODE REVIEW

File: helm-chart/splunk-kubernetes-metrics/values.yaml [hunk 1]
```diff
@@ -109,3 +109,20 @@ tolerations:
...
 affinity: {}
+
+# = Kubernetes Connection Configs =
+kubernetes:
+  # the URL for calling kubernetes API, by default it will be read from the environment variables
+  url:
+  # The port that kubelet is listening to. Default is 10250
+  kubeletPort:
+  # This option is used to get the metrics from summary api on each kubelet using ssl
+  useRestClientSSL:
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-metrics/values.yaml]: this should be true by default

File: helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml [hunk 2]
```diff
@@ -76,7 +95,7 @@ data:
       {{- if or .Values.splunk.hec.caFile .Values.global.splunk.hec.caFile }}
       ca_file /fluentd/etc/splunk/hec_ca_file
       {{- end }}
-      interval {{ or .Values.metricsInterval .Values.global.metricsInterval }}
+      interval 15s
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml]: Restore previous functionality, you removed a configurable.

File: helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml [hunk 3]
```diff
@@ -9,14 +9,33 @@ metadata:
...
-    # system wide configurations
+    # system wide configurations	
     <system>
       log_level {{ or .Values.logLevel .Values.global.logLevel | default "info" }}
     </system>
     <source>
       @type kubernetes_metrics
       tag kube.*
       node_name "#{ENV['SPLUNK_HEC_HOST']}"
+      {{- with $.Values.kubernetes.url }}
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml]: {{- with $.Values.kubernetes.url }} should be {{- with .Values.kubernetes.url }} use consistent syntax throughout the file

File: helm-chart/splunk-kubernetes-metrics/values.yaml [hunk 4]
```diff
@@ -109,3 +109,20 @@ tolerations:
 
 
 affinity: {}
+
+# = Kubernetes Connection Configs =
+kubernetes:
+  # the URL for calling kubernetes API, by default it will be read from the environment variables
+  url:
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-metrics/values.yaml]: Redundant parameter which is not used anywhere, remove this.

File: helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml [hunk 5]
```diff
@@ -9,14 +9,33 @@ metadata:
...
+      kubernetes_url {{ . }}
```
Comment (chaitanyaphalak) [inline helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml]: Redundant parameter which is not used anywhere, remove this.

File: helm-chart/splunk-kubernetes-metrics/values.yaml (hunk 1 above)
Comment (gp510) [reply helm-chart/splunk-kubernetes-metrics/values.yaml]: done

File: helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml (hunk 2 above)
Comment (gp510) [reply helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml]: done

File: helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml (hunk 3 above)
Comment (gp510) [reply helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml]: done

File: helm-chart/splunk-kubernetes-metrics/values.yaml (hunk 4 above)
Comment (gp510) [reply helm-chart/splunk-kubernetes-metrics/values.yaml]: done

File: helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml (hunk 5 above)
Comment (gp510) [reply helm-chart/splunk-kubernetes-metrics/templates/configMap.yaml]: done

**Pull request merged**